### PR TITLE
[ART-4086] gen-assembly: dotnext preview

### PR DESCRIPTION
**Pull request by joepvd**

### CODE REVIEW

**Review by thiagoalessio — APPROVED** [review]
/lgtm

**Review by thiagoalessio — APPROVED** [review]
/lgtm

File: doozerlib/cli/release_gen_assembly.py [hunk 1]
```diff
@@ -73,15 +73,23 @@ def exit_with_error(msg):
...
+        assembly_type = 'preview'
+
     # Calculate previous list
     final_previous_list: Set[VersionInfo] = set()
     if in_flight:
         final_previous_list.add(VersionInfo.parse(in_flight))
     if previous_list:
         final_previous_list |= set(map(VersionInfo.parse, previous_list))
-    elif auto_previous:
+    elif auto_previous and assembly_type not in ['preview', 'custom']:
```
Comment (vfreex) [inline doozerlib/cli/release_gen_assembly.py:90]: AssemblyTypes enum values are preferred.

File: doozerlib/cli/release_gen_assembly.py [hunk 2]
```diff
@@ -73,15 +73,23 @@ def exit_with_error(msg):
     if custom and (auto_previous or previous_list or in_flight):
         exit_with_error("Custom release doesn't have previous list.")
 
+    assembly_type = 'standard'
```
Comment (vfreex) [inline doozerlib/cli/release_gen_assembly.py:76]: Do you mind refactoring this by using enum values defined in AssemblyTypes?

**Review by vfreex — COMMENTED** [review]
I have a suggestion to use AssemblyTypes enumeration values instead of hardcoded assembly type strings. Otherwise LGTM.

**Review by vfreex — APPROVED** [review]
/lgtm

**Pull request merged**